java/code/src/com/suse/oval/OVALCleaner.java

```diff
@@ -0,0 +1,185 @@
+        }
+
+        // Although it's rare, but it's possible to get null criteria trees.
+        root.getDefinitions().removeIf(def -> def.getCriteria() == null);
+
+        root.getDefinitions().forEach(definition -> doCleanupDefinition(definition, osFamily, osVersion));
+        root.getTests().forEach(test -> doCleanupTest(test, osFamily, osVersion));
+        root.getStates().forEach(state -> doCleanupState(state, osFamily, osVersion));
+        root.getObjects().forEach(object -> doCleanupObject(object, osFamily, osVersion));
+    }
+
+    private static void doCleanupDefinition(DefinitionType definition, OsFamily osFamily, String osVersion) {
+        fillCves(definition, osFamily);
+        fillOsFamily(definition, osFamily);
+        fillOsVersion(definition, osVersion);
+
+        if (osFamily == OsFamily.DEBIAN) {
+            convertDebianTestRefs(definition.getCriteria(), osVersion);
+        }
+    }
+
+    private static void fillCves(DefinitionType definition, OsFamily osFamily) {
+        switch (osFamily) {
+            case REDHAT_ENTERPRISE_LINUX:
+            case LEAP:
+            case SUSE_LINUX_ENTERPRISE_SERVER:
+            case SUSE_LINUX_ENTERPRISE_DESKTOP:
+                List<String> cves =
+                        definition.getMetadata().getAdvisory().map(Advisory::getCveList)
+                                .orElse(Collections.emptyList())
+                                .stream().map(AdvisoryCveType::getCve).collect(Collectors.toList());
+                definition.setCves(cves);
+                break;
+            case DEBIAN:
+                definition.setSingleCve(definition.getMetadata().getTitle().split("\\s+")[0]);
+                break;
+            default:
+                throw new NotImplementedException("Cannot extract cve from '" + osFamily + "' OVAL definitions");
+        }
+
+        List<String> cleanCves = definition.getCves().stream().map(OVALCleaner::removeWhitespaceChars)
+                .collect(Collectors.toList());
+
+        definition.setCves(cleanCves);
+    }
+
+    private static String removeWhitespaceChars(String s) {
+        return StringUtils.deleteWhitespace(s);
+    }
+
+    private static void fillOsFamily(DefinitionType definition, OsFamily osFamily) {
+        definition.setOsFamily(osFamily);
+    }
+
+    private static void fillOsVersion(DefinitionType definition, String osVersion) {
+        definition.setOsVersion(osVersion);
+    }
+
+    private static void doCleanupTest(TestType test, OsFamily osFamily, String osVersion) {
+        if (osFamily == OsFamily.DEBIAN) {
+            test.setId(convertDebianId(test.getId(), osVersion));
+            test.setObjectRef(convertDebianId(test.getObjectRef(), osVersion));
+            if (test.getStateRef().isPresent()) {
+                test.setStateRef(convertDebianId(test.getStateRef().get(), osVersion));
+            }
```

Comment:
> SonarCloud: Call "Optional#isPresent()" before accessing the value. 

This is a false positive bug because `test.getStateRef().isPresent()` is being called before calling `test.getStateRef().get()`

Reply:
I rewrote the code using `ifPresent(stateRef -> ... )` and now SonarCloud doesn't detect it as a bug anymore.